bot/handlers.py

```diff
@@ -0,0 +1,201 @@
+import functools
+import random
+
+import psycopg2
+import telebot.types as types
+from environs import Env
+
+env = Env()
+env.read_env()
+
+DB_USER = env.str("DB_USER")
+DB_PASSWORD = env.str("DB_PASSWORD")
+
+
+def cursor(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        conn = psycopg2.connect(database="postgres",
+                                user=DB_USER,
+                                password=DB_PASSWORD,
+                                host="database",
+                                port=5432)
+        curs = conn.cursor()
+        result = func(*args, **kwargs, curs=curs)
+        conn.commit()
+        conn.close()
+        return result
+
+    return wrapper
+
+
+class DBTokensHandler:
+    @staticmethod
+    @cursor
+    def update_me(message: types.Message, curs: psycopg2.extensions.cursor) -> str:
+        """Set new token for User if one exists.
+
+        Args:
+            message: Message object.
+            curs: PostgreSQL cursor object.
+
+        Returns:
+            User token.
+
+        """
+        uid = message.from_user.id
+        token = abs(hash(str(uid - message.id + message.date)))
+        curs.execute("UPDATE users SET username = %s, user_token = %s WHERE user_id = %s",
+                     [
+                           message.from_user.username,
+                           token,
+                           uid
+                       ])
+        return str(token)
+
+    @staticmethod
+    @cursor
+    def set_me(message: types.Message, curs: psycopg2.extensions.cursor) -> str:
+        """Set new token for User if none exists.
+
+        Args:
+            message: Message object.
+            curs: PostgreSQL cursor object.
+
+        Returns:
+            User token.
+
+        """
+        uid = message.from_user.id
+        token = abs(hash(str(uid - message.id + message.date)))
+        curs.execute("INSERT INTO users (username, user_id, user_token) VALUES (%s, %s, %s)",
+                     [
+                           message.from_user.username,
+                           uid,
+                           token
+                       ])
+        return str(token)
+
+    @staticmethod
+    @cursor
+    def get_me(message: types.Message, curs: psycopg2.extensions.cursor) -> str | None:
+        """Get User token from the DB.
+
+        Args:
+            message: Message object.
+            curs: PostgreSQL cursor object.
+
+        Returns:
+            User token if exists, None otherwise.
+
+        """
+        uid = message.from_user.id
+        curs.execute("SELECT user_token FROM users WHERE users.user_id = %s", [uid])
```

Comment:
```suggestion
        curs.execute("SELECT user_token FROM users WHERE users.user_id = %s", (uid, ))
```
по канону всё-таки лучше использовать typle т.к это неизменяемый тип данных